webapp/src/pages/home/Home.tsx

```diff
@@ -1,49 +1,86 @@
     const [ownerFilter, setOwnerFilter] = useState<QueryFilter>();
-    const filters = [authorFilter, organizationFilter, repositoryFilter, ownerFilter];
+    const [starsFilter, setStarsFilter] = useState<QueryFilter>({
+        column: 'rep.stargazers_count',
+        operator: '>=',
+        target: '2'
+    });
+    const [organization, setOrganization] = useState(defaultSelection);
+    const [author, setAuthor] = useState(defaultSelection);
+    const [repositoryOwner, setRepositoryOwner] = useState(defaultSelection);
+    const [repository, setRepository] = useState(defaultSelection);
+    const filters = [authorFilter, organizationFilter, repositoryFilter, ownerFilter, starsFilter];
+    const weekStartsQuery = `
+        SELECT DATE_TRUNC('week', start_date) AS week_start
+        FROM (
+            WITH recursive weeks AS (
+                SELECT CURRENT_DATE - INTERVAL '1 year' + INTERVAL '1 week' AS start_date, 
+                    CURRENT_DATE AS end_date
+                UNION ALL
+                SELECT start_date + INTERVAL '1 week',
+                    end_date
+                FROM weeks
+                WHERE start_date < end_date
+            )
+            SELECT start_date
+            FROM weeks
+        )
+    `;
+    const PullRequestCountByWeekQuery = `
+        SELECT DATE_TRUNC('week', CAST(pr.created_at AS DATE)) AS orderedField,
+               COUNT(DISTINCT title) AS amount
+        FROM main_marts.fct_pull_requests pr
+        LEFT JOIN main_marts.fct_repositories rep ON full_repository_name = rep.full_name
+        ${useQueryFilter([...filters, { column: 'CAST(pr.created_at AS DATE)', operator: '>=', target: 'date_add(CURRENT_DATE(), INTERVAL \'-1 year\')' }])}
+        GROUP BY DATE_TRUNC('week', CAST(pr.created_at AS DATE))
+        ORDER BY orderedField
+    `;
 
-    const authorQuery = 'SELECT distinct author FROM main_marts.fct_pull_requests ORDER BY lower(author);';
-    const organizationQuery = 'SELECT distinct author_organization AS organization FROM main_marts.fct_pull_requests ORDER BY lower(author_organization);';
-    const repositoryQuery = 'SELECT distinct repository FROM main_marts.fct_pull_requests ORDER BY lower(repository);';
-    const ownerQuery = 'SELECT distinct owner FROM main_marts.fct_pull_requests ORDER BY lower(owner);';
-    const pullRequestCountQuery = `SELECT count(*) as amount FROM main_marts.fct_pull_requests ${useQueryFilter(filters)};`;
-    const repoCountQuery = `SELECT count(distinct repository) as amount FROM main_marts.fct_pull_requests ${useQueryFilter(filters)};`;
-    const contributorCountQuery = `SELECT count(distinct author) as amount FROM main_marts.fct_pull_requests ${useQueryFilter(filters)};`;
+    const organizationQuery = 'SELECT distinct pr.author_organization AS organization FROM main_marts.fct_pull_requests pr LEFT JOIN main_marts.fct_repositories rep ON pr.full_repository_name = rep.full_name ORDER BY lower(pr.author_organization);';
```

Comment:
These are starting to get big; we should think what we want to do with this

Reply:
I do agree, created another issue for this: https://github.com/godatadriven/github-contributions/issues/82

Ideally I would like to have some kind of ORM implemented. Currently SQL queries and filtering logic is getting quite big. So I will look into ways to clean this a bit.